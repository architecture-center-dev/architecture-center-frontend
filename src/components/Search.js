import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { TextField, Typography } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    height:'100%'
  },
  containerSearchBar:{
    marginTop: '20%',
    marginBottom: '50px',
  },
  searchBar:{
    backgroundColor: '#ffffff',
    marginTop: '100px'
  }
}));

const keyUpSearch = (event, loadingSolutions) => {
  if(event.keyCode == 13){
     loadingSolutions({
      variables: {
        search: event.target.value
      },
    }); 
  }
}

export default function Search({loadingSolutions}) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Grid container spacing={3} alignContent={'center'} justify={'center'}>
        <Grid item justify="center"  lg={4} xl={4} md={6} xs={12} >

            <div className={classes.containerSearchBar}>
                <Typography variant={'h3'} align={'center'}>Architecture Center</Typography>
                <TextField 
                id="outlined-basic" 
                label="Search by i.e 'kubernetes'" 
                variant="outlined" fullWidth 
                className={classes.searchBar}
                onKeyUp={
                  (event) => keyUpSearch(event, loadingSolutions)
                }/>
            </div>

        </Grid>
      </Grid>
    </div>
  );
}