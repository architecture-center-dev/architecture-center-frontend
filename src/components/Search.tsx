import React, { KeyboardEvent, useEffect, useRef } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { TextField, Typography } from '@material-ui/core';

const useStyles = makeStyles(() => ({
  root: {
    flexGrow: 1,
    height: '100%'
  },
  containerSearchBar: {
    marginTop: '20%',
    marginBottom: '50px',
  },
  searchBar: {
    backgroundColor: '#ffffff',
    marginTop: '100px'
  }
}));

const keyUpSearch = (event: KeyboardEvent, loadingSolutions: Function) => {
  if (event.keyCode == 13) {
    loadingSolutions({
      variables: {
        search: (event.target as HTMLInputElement).value
      },
    });
  }
}

export type SearchProps = {
  loadingSolutions: Function
}

export default function Search({ loadingSolutions }: SearchProps) {
  const classes = useStyles();

  const inputEl = useRef<HTMLInputElement>(null);

  useEffect(() => {
    inputEl?.current?.querySelector('input')?.focus();
  }, [])

  return (
    <div className={classes.root} data-testid="search">
      <Grid container spacing={3} alignContent={'center'} justify={'center'}>
        <Grid item lg={4} xl={4} md={6} xs={12} >

          <div className={classes.containerSearchBar}>
            <Typography variant={'h3'} align={'center'}>Architecture Center</Typography>
            <TextField
              data-testid="searchTextField"
              ref={inputEl}
              id="outlined-basic"
              label="Search by i.e 'kubernetes'"
              variant="outlined" fullWidth
              className={classes.searchBar}
              onKeyUp={
                (event) => keyUpSearch(event, loadingSolutions)
              } />
          </div>

        </Grid>
      </Grid>
    </div>
  );
}