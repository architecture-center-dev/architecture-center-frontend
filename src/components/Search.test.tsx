import * as React from "react";
import Search, { SearchProps } from "./Search";
import { fireEvent, render } from '@testing-library/react'

describe("<Search />", () => {
  it("Component should render correctely", async () => {

    const props: SearchProps = {
      loadingSolutions: jest.fn()
    }

    const { getByTestId, getByText } = render(<Search {...props} />);
    const search = getByTestId('search');

    expect(search).toBeTruthy();
    expect(getByText('Architecture Center')).toBeInTheDocument;
    expect(search).toMatchSnapshot();
  });

  it("When press Enter loadingSolutions is called", async () => {

    const props: SearchProps = {
      loadingSolutions: jest.fn()
    }

    const { getByTestId } = render(<Search {...props} />);
    const searchTextField = getByTestId('searchTextField');

    expect(searchTextField).toBeTruthy();
    fireEvent.keyUp(searchTextField, { value: "some text", keyCode: 13 });

    expect(props.loadingSolutions).toBeCalled();
    expect(searchTextField).toMatchSnapshot();
  });
});
