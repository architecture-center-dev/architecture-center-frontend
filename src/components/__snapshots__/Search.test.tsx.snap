// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Component should render correctely 1`] = `
<div
  class="makeStyles-root-1"
  data-testid="search"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-content-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 MuiGrid-grid-xl-4"
    >
      <div
        class="makeStyles-containerSearchBar-2"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
        >
          Architecture Center
        </h3>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-searchBar-3 MuiFormControl-fullWidth"
          data-testid="searchTextField"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
            data-shrink="true"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Search by i.e 'kubernetes'
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="outlined-basic"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
              >
                <span>
                  Search by i.e 'kubernetes'
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Search /> When press Enter loadingSolutions is called 1`] = `
<div
  class="MuiFormControl-root MuiTextField-root makeStyles-searchBar-10 MuiFormControl-fullWidth"
  data-testid="searchTextField"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
    data-shrink="true"
    for="outlined-basic"
    id="outlined-basic-label"
  >
    Search by i.e 'kubernetes'
  </label>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiOutlinedInput-input"
      id="outlined-basic"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="PrivateNotchedOutline-legendLabelled-13 PrivateNotchedOutline-legendNotched-14"
      >
        <span>
          Search by i.e 'kubernetes'
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
